poc/flp_generic.py

```diff
@@ -640,6 +641,140 @@ def test_vec_set_type_param(self, test_vec):
+    Field = field.Field128
+
+    # Associated parameters
+    GADGETS = None  # Set by `MultiHotHistogram.with_params()`
+    GADGET_CALLS = None  # Set by `MultiHotHistogram.with_params()`
+    MEAS_LEN = None  # Set by `MultiHotHistogram.with_params()`
+    JOINT_RAND_LEN = 2
+    OUTPUT_LEN = None  # Set by `MultiHotHistogram.with_params()`
+
+    def __init__(self, length, max_num_bits, chunk_length):
+        """
+        Instantiate an instance of the `MultiHotHistogram` circuit with the
+        given length, max_num_bits, and chunk_length.
+        """
+        if length <= 0:
+            raise ValueError('invalid length')
+        if max_num_bits <= 0 or max_num_bits >= length:
+            raise ValueError('invalid max_num_bits')
+        if chunk_length <= 0:
+            raise ValueError('invalid chunk_length')
+
+        # Compute the number of bits to represent `max_num_bits`.
+        self.num_bits_for_count = int(math.log2(max_num_bits)) + 1
+        self.offset = self.Field(
+            (1 << self.num_bits_for_count) - 1 - max_num_bits
+        )
+        # Sanity check `offset + length` doesn't overflow field size,
+        # because in validity circuit, we will compute `offset + \sum_i x_i`.
+        if self.Field.MODULUS - self.offset.as_unsigned() <= length:
+            raise ValueError('length and max_num_bits are too large '
+                             'for the current field size')
+
+        self.length = length
+        self.max_num_bits = max_num_bits
+        self.chunk_length = chunk_length
+        self.GADGETS = [ParallelSum(Mul(), chunk_length)]
+        # The number of bit entries are `length + num_bits_for_count`,
+        # so the number of gadget calls is equal to
+        # `ceil((length + num_bits_for_count) / chunk_length)`.
+        self.GADGET_CALLS = [
+            (length + self.num_bits_for_count + chunk_length - 1) //
+            chunk_length
+        ]
+        self.MEAS_LEN = self.length + self.num_bits_for_count
+        self.OUTPUT_LEN = self.length
+
+    def eval(self, meas, joint_rand, num_shares):
+        self.check_valid_eval(meas, joint_rand)
+
+        # Check that each bucket is one or zero.
+        range_check = self.Field(0)
+        r = joint_rand[0]
+        r_power = r
+        shares_inv = self.Field(num_shares).inv()
+        for i in range(self.GADGET_CALLS[0]):
+            inputs = [None] * (2 * self.chunk_length)
+            for j in range(self.chunk_length):
+                index = i * self.chunk_length + j
+                if index < len(meas):
+                    meas_elem = meas[index]
+                else:
+                    meas_elem = self.Field(0)
+
+                inputs[j * 2] = r_power * meas_elem
+                inputs[j * 2 + 1] = meas_elem - shares_inv
+
+                r_power *= r
+
+            range_check += self.GADGETS[0].eval(self.Field, inputs)
+
+        # Check that `offset + \sum_{i=0}^{length-1} meas[i]` is equal to the
```

Comment:
Prefer simpler language in comments
```suggestion
        # Check that `offset` plus the some of the buckets is equal to the
```